inference_segmentation.py

```diff
@@ -225,26 +227,22 @@ def segmentation(param,
     del fp
 
     fp = np.memmap(tp_mem, dtype='float16', mode='r', shape=(tf_len, h_padded, w_padded, num_classes))
-    pred_img = np.zeros((h_padded, w_padded), dtype=np.uint8)
+    pred_heatmap = np.zeros((h_padded, w_padded, num_classes), dtype=np.float16)
     for row, col in tqdm(itertools.product(range(0, input_image.height, chunk_size),
                                            range(0, input_image.width, chunk_size)),
                          leave=False, total=total_inf_windows, desc="Writing to array"):
         arr1 = (fp[:, row:row + chunk_size, col:col + chunk_size, :]).max(axis=0)
         if single_class_mode:
             arr1 = sigmoid(arr1)
-            arr1 = (arr1 > threshold)
-            arr1 = np.squeeze(arr1, axis=2).astype(np.uint8)
-        else:
-            arr1 = np.argmax(arr1, axis=-1).astype(np.uint8)
-        pred_img[row:row + chunk_size, col:col + chunk_size] = arr1
+        pred_heatmap[row:row + chunk_size, col:col + chunk_size, :] = arr1
```

Comment:
the name change pred_img --> pred_heatmap is not consistent with the output of the segmentation function.

Reply:
Can the "segmentation function" become a function that returns a heatmap rather than final prediction itself? I'd add this in docstrings, but how is pred_heatmap not consistent with the ouput of this function if we decide that it now returns a heatmap?

Reply:
Yes, it is fine. I understand now that thresholding has been moved to 'class_from_heatmap',